lib/http2/connection.c

```diff
@@ -867,7 +867,9 @@ static int handle_goaway_frame(h2o_http2_conn_t *conn, h2o_http2_frame_t *frame,
     if ((ret = h2o_http2_decode_goaway_payload(&payload, frame, err_desc)) != 0)
         return ret;
 
-    /* nothing to do, since we do not open new streams by ourselves */
+    /* stop opening new push streams hereafter */
+    conn->push_stream_ids.max_open = INT32_MAX;
```

Comment:
Could we use `0x7ffffffe` instead, since `INT32_MAX` is an ID of a _pull_ stream?

Other than that the PR seems ready for merge. Thank you for working on the fix.

Reply:
Thank you for the comment. Fixed at https://github.com/h2o/h2o/pull/1555/commits/0bbc06282b849a7dfa7c53c63d888a41aa8a84bd

Reply:
@kazuho done in https://github.com/h2o/h2o/pull/1555/commits/0bbc06282b849a7dfa7c53c63d888a41aa8a84bd